### PR TITLE
Implement WithJujuAttrs on the charm store repo.

### PR DESCRIPTION
**Pull request by frankban**

### CODE REVIEW

File: charmrepo/charmstore.go [hunk 1]
```diff
@@ -226,3 +226,19 @@ func (s *CharmStore) WithTestMode() Interface {
 	newRepo.client.DisableStats()
 	return &newRepo
 }
+
+// JujuMetadataHTTPHeader is the HTTP header name used to send Juju metadata
```
Comment (urosj) [inline charmrepo/charmstore.go]: Will this really be any juju or just JES?

File: charmrepo/charmstore_test.go [hunk 2]
```diff
@@ -238,6 +239,43 @@ func (s *charmStoreRepoSuite) TestGetWithTestMode(c *gc.C) {
...
+	})
+
+	// Make a first request without Juju attrs.
+	_, err := repo.Get(url)
+	c.Assert(err, jc.ErrorIsNil)
+	c.Assert(header.Get(charmrepo.JujuMetadataHTTPHeader), gc.Equals, "")
+
+	// Make a second request after setting Juju attrs.
+	repo = repo.(*charmrepo.CharmStore).WithJujuAttrs(map[string]string{
+		"k1": "v1", "k2": "v2",
```
Comment (mhilton) [inline charmrepo/charmstore_test.go]: as this is on multiple lines anyway you might as well have one key/value per line.

**Comment by mhilton** [comment]
:+1:

**Comment by urosj** [comment]
:+1:

File: charmrepo/charmstore_test.go (hunk 2 above)
Comment (frankban) [inline charmrepo/charmstore_test.go]: Done.

**Comment by frankban** [comment]
Thanks for the reviews!

**Pull request merged**